package/MDAnalysis/coordinates/chain.py

```diff
@@ -79,81 +79,105 @@ class ChainReader(base.ProtoReader):
+                                 for filename in filenames], dtype=object)
+        self.filenames = np.array([fn[0] if isinstance(fn, tuple) else fn for fn in filenames])
         # pointer to "active" trajectory index into self.readers
         self.__active_reader_index = 0
 
-        self.skip = kwargs.get('skip', 1)
+        self.skip = skip
         self.n_atoms = self._get_same('n_atoms')
-        #self.fixed = self._get_same('fixed')
 
         # Translation between virtual frames and frames in individual
-        # trajectories.
-        # Assumes that individual trajectories i contain frames that can
-        # be addressed with an index 0 <= f < n_frames[i]
+        # trajectories. Assumes that individual trajectories i contain frames
+        # that can be addressed with an index 0 <= f < n_frames[i]
 
-        # Build a map of frames: ordered list of starting virtual
-        # frames; the index i into this list corresponds to the index
-        # into self.readers
+        # Build a map of frames: ordered list of starting virtual frames; the
+        # index i into this list corresponds to the index into self.readers
         #
         # For virtual frame 0 <= k < sum(n_frames) find corresponding
-        # trajectory i and local frame f (i.e. readers[i][f] will
-        # correspond to ChainReader[k]).
-
+        # trajectory i and local frame f (i.e. readers[i][f] will correspond to
+        # ChainReader[k]).
         # build map 'start_frames', which is used by _get_local_frame()
-        n_frames = self._get('n_frames')
+        n_frames = np.array(self._get('n_frames'))
         # [0]: frames are 0-indexed internally
         # (see Timestep.check_slice_indices())
         self.__start_frames = np.cumsum([0] + n_frames)
 
         self.n_frames = np.sum(n_frames)
-        self.dts = np.array(self._get('dt'))
+        self.dts = np.array([dt if dt is not None else 0 for dt in self._get('dt')])
         self.total_times = self.dts * n_frames
 
         #: source for trajectories frame (fakes trajectory)
         self.__chained_trajectories_iter = None
 
+        # calculate new start_frames to have a time continuous trajectory.
+        if continuous:
+            # TODO: check for some filetype!
+            # TODO: allow floating point precision in dt check
+            dt = self._get_same('dt')
+            # sort
+            sort_idx = np.argsort([r.ts.time for r in self.readers])
+            self.readers = self.readers[sort_idx]
+            self.filenames = self.filenames[sort_idx]
+            self.total_times = self.dts * n_frames[sort_idx]
+            # rebuild lookup table
+            sf = [0, ]
+            n_frames = 0
+            for r1, r2 in zip(self.readers[:-1], self.readers[1:]):
+                r2[0], r1[0]
+                start_time = r2.time
+                # find end where trajectory was restarted from
+                for ts in r1[::-1]:
```

Comment:
I expect this to be rather slow. Trying first on the basis of the first time, the number of frames, and `dt` should save a lot of time for the general case. Iterating over the frames would then be the fall-back if the fast heuristics did not work.

Reply:
yeah I would expect this to. But I rather first have this simple algorithm and get it correct with tests for corner cases. We can still make it faster at the end of the PR.